Sources/Core/AWSClientRuntime/Errors/AWSHttpServiceError.swift

```diff
@@ -9,4 +9,20 @@ public let X_AMZN_REQUEST_ID_HEADER = "X-Amzn-RequestId"
 
 public protocol AWSHttpServiceError: HttpServiceError {
     var _requestID: String? { get set }
+
+    /// For an error that is identifiable to a Smithy error type, the error code will be the
+    /// non-namespaced name of the error shape.  For unidentified errors, the error code
+    /// may be a Smithy type not handled by this SDK, or may contain other identifying
+    /// information.
```

Comment:
imo we should avoid using terms like `smithy` in our code base to future proof against smithy being renamed or replaced.

```suggestion
    /// The non-namespaced name of the error. This value is provided by the operation response and is used to map the response into a specific error. If no error model exists with this name, then an `Unknown` error is created.
```